packages/devtools/lib/src/timeline/frame_flame_chart.dart

```diff
@@ -407,3 +430,142 @@ class FlameChartItem {
         colorToCss(selected ? selectedTextColor : defaultTextColor);
   }
 }
+
+class TimelineGrid extends CoreElement {
+  TimelineGrid(this._frameDurationMs, this.flameChartWidth)
+      : super('div', classes: 'flame-chart-grid') {
+    flex();
+    layoutHorizontal();
+    _drawGrid(baseGridInterval);
+  }
+
+  static const baseGridInterval = 150;
+
+  final num _frameDurationMs;
+
+  num zoomLevel = 1;
+
+  num flameChartWidth;
+
+  num get flameChartWidthWithInsets => flameChartWidth + 2 * flameChartInset;
+
+  num interval = baseGridInterval;
+
+  final List<TimelineGridItem> _gridItems = [];
+
+  void _drawGrid(num interval) {
+    // Draw the first grid item since it will have a different width than the
+    // rest.
+    final gridItem = TimelineGridItem(0, flameChartInset, 0);
```

Comment:
nit:
is GridLine clearer than GridItem?

Each item is concretely a vertical GridLine with a label.

Reply:
Technically each grid item is a container for a line and a label. I didn't want to call it GridLine because it's not really just a line. We modify it's left/width values, and those left/width values don't correspond to the literal line, they correspond to the container that contains the line.